### PR TITLE
Add plugin for moonrepo (moonrepo.dev)

### PR DESCRIPTION
**Pull request by ethanjdiamond**

### CODE REVIEW

File: packages/knip/src/plugins/moonrepo/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+// link to moonrepo docs: https://moonrepo.dev/docs
+
+const title = 'moonrepo';
+
+const enablers: EnablerPatterns = ['@moonrepo/cli'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config: string[] = ['**/moon.yml', '.moon/tasks.yml', '.moon/tasks/*.yml'];
```
[the remark this answers is not in the document]
Comment (ethanjdiamond) [reply packages/knip/src/plugins/moonrepo/index.ts:14]: I didn't know how to test this where there's various "workspaces", as I'm trying to simulate a monorepo. So the .moon/tasks.yml and .moon/tasks/*.yml are going to be against the workspace root, while moon.yml lives in the project (package) root.  How would I test that?

File: packages/knip/test/plugins/moonrepo.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ...baseArguments,
+    cwd,
+  });
+
+  assert(issues.binaries['.moon/tasks.yml']['ls-lint']);
+  assert(issues.binaries['.moon/tasks/typescript.yml']['tsc']);
+  assert(issues.binaries['.moon/tasks/typescript.yml']['eslint']);
+  assert(issues.binaries['apps/a/moon.yml']['vite-node']);
+
+  assert(issues.files.has(join(cwd, 'tools/linters/lint-readme.ts')));
```
[the remark this answers is not in the document]
Comment (ethanjdiamond) [reply packages/knip/test/plugins/moonrepo.test.ts:21]: Ya I understand what you're going for. Let me see what I can do.

File: packages/knip/src/plugins/moonrepo/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ethanjdiamond) [reply packages/knip/src/plugins/moonrepo/index.ts:14]: nvm figured it out

File: packages/knip/test/plugins/moonrepo.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (webpro) [reply packages/knip/test/plugins/moonrepo.test.ts:27]: The solution looks correct, but I'm still seeing the file unused? What I usually do to debug is something like `npx tsx ../../src/cli.ts --debug` from inside the fixtures folder, that should show some insights into what the plugin returns (again, I'm happy to take over, but of course it's cool if you could wrap this up).

[the remark this answers is not in the document]
Comment (ethanjdiamond) [reply packages/knip/test/plugins/moonrepo.test.ts:27]: So I played around with it a bunch and I'm kind of stumped. Is there something funky going on with the runtime? 

When I logged the `command` on [this line](https://github.com/webpro/knip/blob/aab7fa2640e81955afe5b32894efbd88da62c8ce/packages/knip/src/plugins/moonrepo/index.ts#L23), it's correct. When I go into `getDependenciesFromScripts` and log `npmScripts` on [literally the first line](https://github.com/webpro/knip/blob/aab7fa2640e81955afe5b32894efbd88da62c8ce/packages/knip/src/binaries/index.ts#L11), it's showing an empty array. 

Any idea what's going on there?

[the remark this answers is not in the document]
Comment (webpro) [reply packages/knip/test/plugins/moonrepo.test.ts]: - The tests are "end-to-end" so the first invocation is likely from `package.json`.
- I saw a few things were not in place at all, so I just went ahead and added those in this PR (and will clean up later after merging)

**Comment by webpro** [comment]
If you're OK with this, I'll merge it!

**Comment by webpro** [comment]
Odd, those commits after rebase, no worries about that btw.

**Comment by ethanjdiamond** [comment]
Yup, merge away! Thanks.

**Comment by webpro** [comment]
:rocket: _This pull request is included in v5.9.0. See [Release 5.9.0](https://github.com/webpro/knip/releases/tag/5.9.0) for release notes._

_Using Knip in a commercial project? Please consider [sponsoring me](https://github.com/sponsors/webpro)._

**Pull request merged**